### PR TITLE
Provide more visibility into BPF filesystem mounting

### PR DESCRIPTION
**Pull request by tgraf**
Related: #2924

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

File: pkg/bpf/bpffs.go [hunk 1]
```diff
@@ -210,12 +197,22 @@ func mountFS() error {
...
+		return fmt.Errorf("BPF filesystem path %s is not mounted", mapRoot)
+	}
+
+	// This is the value of the BPF Filesystem defined in
+	// uapi/linux/magic.h The magic value can potentially be misleading if
+	// the BPF filesystem is mounted in the host and then volume mapped
+	// into a container.
+	magic := uint32(0xCAFE4A11)
+	if uint32(fsdata.Type) != magic {
+		log.WithField(logfields.Path, mapRoot).Warning("BPF root is not a BPF filesystem (%d != %d)",
```
Comment (manalibhutiyani) [inline pkg/bpf/bpffs.go]: Should this be a `Log.Fatal` instead, now that we can detect that the mounted BPF file system is not of kind BPF?

Comment (manalibhutiyani) [reply pkg/bpf/bpffs.go]: Actually how is this different from the `IsBpffs` logic we had before? I am a bit confused?

Comment (tgraf) [reply pkg/bpf/bpffs.go]: It was previously casting to `int32` while `0xCAFE4A11` requires an `uint32`

```
-	magic := uint32(0xCAFE4A11)
-	var fsdata unix.Statfs_t
-	if err := unix.Statfs(path, &fsdata); err != nil {
-		log.WithField(logfields.Path, path).Error("BPF filesystem path is not mounted")
-		return false
-	}
-	return int32(magic) == int32(fsdata.Type)
```

I'll push another commit which changes this to `stat` and we can make it fail hard as it works in container environments.

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
I've attempted to use `stat(1)` to verify the filesystem but ran into the same issues as we've seen before. It worked from the shell when I invoked `kubectl exec -ti cilium-xxx bash` and ran `stat /sys/fs/bpf` but when attempting the same from code right after mounting, it never succeeds and the filesystem type reported is `sysfs` instead of `bpf_fs`

**Review by ianvernon — APPROVED** [review]

**Review by joestringer — APPROVED** [review]
When it's volume-mapped into the container can we determine this based on the magic value and attempt to find the underlying FS?

Changes LGTM otherwise.

**Comment by tgraf** [comment]
> When it's volume-mapped into the container can we determine this based on the magic value and attempt to find the underlying FS?

It says `sysfs`, the weird part is that it will start reporting `bpf_fs` after a while. It only seems to report `sysfs` immediately after mounting.

**Pull request merged**